lib/improver/spotdata/read_input.py

```diff
@@ -78,7 +81,12 @@ def process(self, filepath, diagnostic):
         An iris.cube.Cube containing the data from the netCDF file.
 
         """
-        function = getattr(self, self.method)
+        try:
+            function = getattr(self, self.method)
+        except:
+            raise AttributeError('Unknown method "{}" passed to {}.'.format(
+                self.method, self.__class__.__name__))
+
         return function(filepath, diagnostic)
 
     @staticmethod
```

Comment:
Perhaps, but I don't wish to pass method '_single_file' and I don't want to misalign the call and the function name.